### PR TITLE
RetryableError should not block reconciliation attempts

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind bug

**What this PR does / why we need it**:
This PR will allow retrying the reconciliation, after an unsuccessful attempt would return an error code of `RetryableError` from the infrastructure. Previously the `RetryableError` marked as infrastructure dependency error would not be reconciled.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Infrastructure dependency errors containing the `RetryableError` will not stop automatic reconciliation attempts.
```

### CODE REVIEW

**Review by rfranzke — DISMISSED** [review]
/lgtm
/assign @ialidzhikov

File: pkg/apis/core/v1beta1/types_common.go [hunk 1]
```diff
@@ -32,6 +32,8 @@ const (
 	// ErrorInfraDependencies indicates that the last error occurred due to dependent objects on the infrastructure level.
 	// It is classified as a non-retryable error code.
 	ErrorInfraDependencies ErrorCode = "ERR_INFRA_DEPENDENCIES"
+	// ErrorRetryableInfraDependencies indicates that the last error occurred due to dependent objects on the infrastructure level, but operation should be retried.
+	ErrorRetryableInfraDependencies ErrorCode = "ERR_RETRYABLE_INFRA_DEPENDENCIES"
```
Comment (ialidzhikov) [inline pkg/apis/core/v1beta1/types_common.go:36]: Can you update https://github.com/gardener/gardener/blob/master/docs/usage/shoot_status.md#error-codes?
Can you add the error code also for the internal type and v1alpha1 - https://github.com/gardener/gardener/blob/f86f6e4ce3ff57545551e0331411e654aa7da33f/pkg/apis/core/types_common.go#L22 and https://github.com/gardener/gardener/blob/f86f6e4ce3ff57545551e0331411e654aa7da33f/pkg/apis/core/v1alpha1/types_common.go#L20?

**Review by ialidzhikov — CHANGES_REQUESTED** [review]

File: docs/usage/shoot_status.md [hunk 2]
```diff
@@ -99,6 +99,7 @@ Known error codes are:
 - `ERR_INFRA_INSUFFICIENT_PRIVILEGES` - indicates that the last error occurred due to insufficient infrastructure privileges. It is classified as a non-retryable error code.
 - `ERR_INFRA_QUOTA_EXCEEDED` - indicates that the last error occurred due to infrastructure quota limits. It is classified as a non-retryable error code.
 - `ERR_INFRA_DEPENDENCIES` - indicates that the last error occurred due to dependent objects on the infrastructure level. It is classified as a non-retryable error code.
+- `ERR_RETRYABLE_INFRA_DEPENDENCIES` - indicates that the last error occurred due to dependent objects on  on the infrastructure level. "Retryable" means that the occurred error is likely to be resolved in a ungraceful manner after given period of time.
```
Comment (ialidzhikov) [inline docs/usage/shoot_status.md]: This statement about "retryable" with `is likely to be resolved in a ungraceful manner after given period of time` is applicable for `ERR_RETRYABLE_CONFIGURATION_PROBLEM`. I guess there are no `ungraceful manners` in the `ERR_RETRYABLE_INFRA_DEPENDENCIES`, so le't reword this to avoid confusion.

```suggestion
- `ERR_RETRYABLE_INFRA_DEPENDENCIES` - indicates that the last error occurred due to dependent objects on the infrastructure level, but the operation should be retried.
```

**Review by ialidzhikov — APPROVED** [review]
Thanks!

/lgtm
